### ISSUE TITLE
Cypress 9.2.0 hangs under Linux + Docker using cypress/base:16.13.0 + Electron; possibly video/screenshot related

### ISSUE THREAD

**Issue by xeger** [cypress-io/cypress#19617]
### Current behavior

I have a resource-intensive test suite that uses Cucumber plus numerous plugins. If I upgrade from 8.3 to 9.2, my test suite begins hanging after running a few test cases. The hang seems related to video; if I disable video recording in `cypress.json` then the hang converts into a `SIGSEGV` crash the first time Cypress takes a screenshot.

**Workaround**: if I switch my base image to `cypress/browsers:node16.5.0-chrome94-ff93` and run my test suite with `--browser chrome` then everything works well.

I have collected some diagnostic (see "Other") which may be of use in troubleshooting.

In the past, Cypress support has suggested that hangs are due to various `afterEach` hooks that I am running. I can reproduce this hang even if I comment out _all_ of my after-hooks and disable numerous plugins (such as `cypress-terminal-logs`) so I am fairly confident that the hang is _not_ related to hooks, this time. (For one thing, Chrome should hang too if this were the case!)

I will run the suite with all hooks disabled and amend my "Other" with a server debug trace that has less hooks, to prove my point.

### Desired behavior

Ideal behavior for Cypress would be to run to completion, and not to hang or crash. ;-)

### Test code to reproduce

Unable to provide open-source reproduction at this time (my suite setup is elaborate and complex). I will try to isolate, but because the runtime configuration (Docker, etc) is clearly a factor, it may be difficult.

### Cypress Version

9.2.0

### Other

One potential clue is this warning that I see printed periodically with Electron, but not with other browsers:
```
[30:1230/001147.678227:ERROR:bus.cc(392)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
```

If I enable `DEBUG` logging in the server process, some other potential clues come to light; for instance the hang _may_ be related to an uncaught exception or top-level promise rejection, because I can see (via `cypress-terminal-logs`) that my app is making a Sentry report shortly before the hang happens:

```
# this happens in the middle of a test
      cy:command ✘  assert	expected **a,button,card-header:component-text(/^Add User[\\s\\W]*$/i)** to be **visible**


# evidence that perhaps my app experiences an error
        cy:fetch ➟  POST https://REDACTED.ingest.sentry.io/api/REDACTED/envelope/?sentry_key=REDACTED&sentry_version=7
                    Status: 200

# evidence of my app's after-hook running
      cy:command ✔  window	
      cy:request ✔  POST http://global:3000/oauth2/revoke
                    Status: 200
                    Response body: {}
      cy:command ✔  clearCookies	
  cypress:server:plugins promise resolved for id 'inv356' with value null +20s
  cypress:server:task result is: null +30ms
```

After the Sentry report, I see more of my app's after-hooks running, with more evidence that my application has crashed -- a command to find a button fails
which would happen if my app displayed its full-page crash handler after catching an error.

What I _don't_ understand is why Cypress decides to end the suite and "end" the run after that failure (I assume that is what the `suite end` and `end` events means).

Moreover, I don't understand why this crash only reproduces on Electron.

```
  cypress:server:project onMocha hook +411ms
  cypress:server:reporter got mocha event 'hook' with args: [ { id: 'r7', title: '"after all" hook: afterAll', hookName: 'after all', hookId: 'h4', body: 'function afterAll() {\n' + '    writeCoverage(state.getInteractions());\n' + '    state.reset();\n' + '}', type: 'hook', file: 'cypress/integration/firm/company_settings/user_management/create.feature', invocationDetails: { function: 'Object.eval', fileUrl: 'http://api:3000/__cypress/tests?p=cypress/support/index.ts', originalFile: 'http://api:3000/__cypress/tests?p=cypress/support/index.ts', line: 9566, column: 2, whitespace: '    ', stack: 'Error\n' + '    at Object.eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:9566:1)\n' + '    at __webpack_require__ (http://api:3000/__cypress/tests?p=cypress/support/index.ts:20:30)\n' + '    at Object.eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:9501:18)\n' + '    at __webpack_require__ (http://api:3000/__cypress/tests?p=cypress/support/index.ts:20:30)\n' + '    at eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:84:18)\n' + '    at eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:87:10)\n' + '    at eval (<anonymous>)' }, currentRetry: 0, retries: -1, _slow: 10000 } ] +412ms
  cypress:server:project onMocha hook end +3ms
  cypress:server:reporter got mocha event 'hook end' with args: [ { id: 'r7', title: '"after all" hook: afterAll', hookName: 'after all', hookId: 'h4', body: 'function afterAll() {\n' + '    writeCoverage(state.getInteractions());\n' + '    state.reset();\n' + '}', type: 'hook', duration: 2, file: 'cypress/integration/firm/company_settings/user_management/create.feature', invocationDetails: { function: 'Object.eval', fileUrl: 'http://api:3000/__cypress/tests?p=cypress/support/index.ts', originalFile: 'http://api:3000/__cypress/tests?p=cypress/support/index.ts', line: 9566, column: 2, whitespace: '    ', stack: 'Error\n' + '    at Object.eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:9566:1)\n' + '    at __webpack_require__ (http://api:3000/__cypress/tests?p=cypress/support/index.ts:20:30)\n' + '    at Object.eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:9501:18)\n' + '    at __webpack_require__ (http://api:3000/__cypress/tests?p=cypress/support/index.ts:20:30)\n' + '    at eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:84:18)\n' + '    at eval (http://api:3000/__cypress/tests?p=cypress/support/index.ts:87:10)\n' + '    at eval (<anonymous>)' }, currentRetry: 0, retries: -1, _slow: 10000 } ] +2ms
  cypress:server:project onMocha test end +3ms
  cypress:server:reporter got mocha event 'test end' with args: [ { _testConfig: { testConfigList: [], unverifiedTestConfig: {} }, id: 'r7', title: 'Invite a User with Limited Access', err: { message: 'Timed out retrying after 8000ms: Expected to find element: `a,button,card-header:component-text(/^Add User[\\s\\W]*$/i)`, but never found it.', type: 'existence', name: 'AssertionError', stack: 'AssertionError: Timed out retrying after 8000ms: Expected to find element: `a,button,card-header:component-text(/^Add User[\\s\\W]*$/i)`, but never found it.\n' + '    at clickElement (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:52126:112)\n' + '    at Context.eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:19067:57)', sourceMappedStack: 'AssertionError: Timed out retrying after 8000ms: Expected to find element: `a,button,card-header:component-text(/^Add User[\\s\\W]*$/i)`, but never found it.\n' + '    at clickElement (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:52126:113)\n' + '    at Context.eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:19067:58)', parsedStack: [Array] }, state: 'failed', body: 'function () {\n' + '      const state = window.testState;\n' + '      return cy\n' + '        .then(() => state.onStartScenario(scenario, indexedSteps))\n' + '        .then(() =>\n' + '          resolveAndRunBeforeHooks.call(this, scenario.tags, state.feature.name)\n' + '        )\n' + '        .then(() =>\n' + '          indexedSteps.forEach((step) =>\n' + '            stepTest.call(this, state, step, rowData)\n' + '          )\n' + '        )\n' + '        .then(() => state.onFinishScenario(scenario));\n' + '    }', type: 'test', duration: 17937, wallClockStartedAt: '2021-12-30T00:17:02.642Z', timings: { lifecycle: 859, 'before each': [Array], test: [Object], 'after each': [Array], 'after all': [Array] }, file: null, final: true, currentRetry: 2, retries: 2, _slow: 10000 } ] +3ms
  cypress:server:project onMocha suite end +7ms
  cypress:server:reporter got mocha event 'suite end' with args: [ { id: 'r2', title: 'Invite Users', root: false, type: 'suite', file: null, invocationDetails: { function: 'Object.eval', fileUrl: 'http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature', originalFile: 'http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature', line: 39087, column: 4, whitespace: '    ', stack: 'Error\n' + '    at Object.eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:39087:3)\n' + '    at __webpack_require__ (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:20:30)\n' + '    at Object.eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:39053:18)\n' + '    at __webpack_require__ (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:20:30)\n' + '    at eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:84:18)\n' + '    at eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:87:10)\n' + '    at eval (<anonymous>)' }, retries: -1, _slow: 10000 } ] +7ms
  cypress:server:project onMocha test:after:run [34m+1ms
  cypress:server:reporter got mocha event 'test:after:run' with args: [ { _testConfig: { testConfigList: [], unverifiedTestConfig: {} }, id: 'r7', title: 'Invite a User with Limited Access', err: { message: 'Timed out retrying after 8000ms: Expected to find element: `a,button,card-header:component-text(/^Add User[\\s\\W]*$/i)`, but never found it.', type: 'existence', name: 'AssertionError', stack: 'AssertionError: Timed out retrying after 8000ms: Expected to find element: `a,button,card-header:component-text(/^Add User[\\s\\W]*$/i)`, but never found it.\n' + '    at clickElement (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:52126:112)\n' + '    at Context.eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:19067:57)', sourceMappedStack: 'AssertionError: Timed out retrying after 8000ms: Expected to find element: `a,button,card-header:component-text(/^Add User[\\s\\W]*$/i)`, but never found it.\n' + '    at clickElement (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:52126:113)\n' + '    at Context.eval (http://api:3000/__cypress/tests?p=cypress/integration/firm/company_settings/user_management/create.feature:19067:58)', parsedStack: [Array] }, state: 'failed', body: 'function () {\n' + '      const state = window.testState;\n' + '      return cy\n' + '        .then(() => state.onStartScenario(scenario, indexedSteps))\n' + '        .then(() =>\n' + '          resolveAndRunBeforeHooks.call(this, scenario.tags, state.feature.name)\n' + '        )\n' + '        .then(() =>\n' + '          indexedSteps.forEach((step) =>\n' + '            stepTest.call(this, state, step, rowData)\n' + '          )\n' + '        )\n' + '        .then(() => state.onFinishScenario(scenario));\n' + '    }', type: 'test', duration: 17937, wallClockStartedAt: '2021-12-30T00:17:02.642Z', wallClockDuration: 19946, timings: { lifecycle: 859, 'before each': [Array], test: [Object], 'after each': [Array], 'after all': [Array] }, file: null, final: true, currentRetry: 2, retries: 2, _slow: 10000 } ] +1ms
  cypress:server:project onMocha suite end +2ms
  cypress:server:reporter got mocha event 'suite end' with args: [ { id: 'r1', title: '', root: true, type: 'suite', file: 'cypress/integration/firm/company_settings/user_management/create.feature', retries: -1, _slow: 10000 } ] +2ms
  cypress:server:project onMocha end +1ms
  cypress:server:reporter got mocha event 'end' with args: [ { end: '2021-12-30T00:17:22.641Z' } ] +1ms
  cypress:server:browsers:electron debugger: sending Page.screencastFrameAck with params { sessionId: 1 } +279ms
  cypress:server:browsers:electron debugger: received response to Page.screencastFrameAck: {} +2ms
  cypress:server:browsers:electron debugger: sending Page.screencastFrameAck with params { sessionId: 1 } +55ms
  cypress:server:browsers:electron debugger: received response to Page.screencastFrameAck: {} +1ms
  cypress:server:browsers:electron debugger: sending Page.screencastFrameAck with params { sessionId: 1 } +14ms
  cypress:server:browsers:electron debugger: received response to Page.screencastFrameAck: {} +1ms
  cypress:server:util:process_profiler current & mean memory and CPU usage by process group:
  cypress:server:util:process_profiler ┌─────────┬───────────────────┬──────────────┬────────────────────┬────────────┬────────────────┬──────────┬──────────────┬─────────────┐
  cypress:server:util:process_profiler │ (index) │       group       │ processCount │        pids        │ cpuPercent │ meanCpuPercent │ memRssMb │ meanMemRssMb │ maxMemRssMb │
  cypress:server:util:process_profiler ├─────────┼───────────────────┼──────────────┼────────────────────┼────────────┼────────────────┼──────────┼──────────────┼─────────────┤
  cypress:server:util:process_profiler │    0    │    'Electron'     │      1       │       '256'        │   19.08    │     15.89      │  373.65  │    278.85    │   373.65    │
  cypress:server:util:process_profiler │    1    │ 'electron-shared' │      4       │ '36, 192, 37, 213' │   16.49    │     13.73      │  254.15  │    245.68    │   260.08    │
  cypress:server:util:process_profiler │    2    │     'cypress'     │      1       │        '30'        │   55.27    │     47.89      │  248.22  │    243.28    │   255.86    │
  cypress:server:util:process_profiler │    3    │     'plugin'      │      1       │       '238'        │    0.05    │      2.53      │  146.84  │    171.95    │    241.1    │
  cypress:server:util:process_profiler │    4    │     'ffmpeg'      │      1       │       '251'        │    1.25    │      1.2       │  76.43   │    72.46     │    76.43    │
  cypress:server:util:process_profiler │    5    │      'other'      │      2       │    '999, 1000'     │     0      │      0.03      │   3.4    │     3.39     │    3.45     │
  cypress:server:util:process_profiler │    6    │      'TOTAL'      │      10      │        '-'         │   92.14    │     80.68      │  1102.7  │    999.74    │   1106.26   │
  cypress:server:util:process_profiler └─────────┴───────────────────┴──────────────┴────────────────────┴────────────┴────────────────┴──────────┴──────────────┴─────────────┘ +10s
```

After that, it's just thousands of lines of process profiling and screenshots for 10 minutes until my CI system times out the job.

**Comment by waltz** [comment]
I'm seeing something very similar. I haven't been able to track it down to a specific test file. I see a few `MaxListenersExceeded` warnings before this though other GH issues say that's not related. I'm stumped on this one.

```
cypress:server:video capture stderr log { message: 'frame=19146 fps= 25 q=12.0 size=   21248kB time=00:12:44.88 bitrate= 227.6kbits/s dup=9961 drop=0 speed=1.01x    ' } +583ms
  cypress:server:video capture stderr log { message: 'frame=19159 fps= 25 q=12.0 size=   21248kB time=00:12:45.40 bitrate= 227.4kbits/s dup=9968 drop=0 speed=1.01x    ' } +500ms
  cypress:server:video capture stderr log { message: 'frame=19173 fps= 25 q=12.0 size=   21248kB time=00:12:45.96 bitrate= 227.2kbits/s dup=9975 drop=0 speed=1.01x    ' } +582ms
  cypress:server:video capture stderr log { message: 'frame=19186 fps= 25 q=12.0 size=   21248kB time=00:12:46.48 bitrate= 227.1kbits/s dup=9982 drop=0 speed=1.01x    ' } +500ms
  cypress:server:video capture stderr log { message: 'frame=19200 fps= 25 q=12.0 size=   21248kB time=00:12:47.04 bitrate= 226.9kbits/s dup=9989 drop=0 speed=1.01x    ' } +584ms
  cypress:server:video capture stderr log { message: 'frame=19215 fps= 25 q=12.0 size=   21248kB time=00:12:47.64 bitrate= 226.8kbits/s dup=9997 drop=0 speed=1.01x    ' } +583ms
  cypress:server:video capture stderr log { message: 'More than 10000 frames duplicated' } +333ms
  cypress:server:video capture stderr log { message: 'frame=19229 fps= 25 q=12.0 size=   21248kB time=00:12:48.20 bitrate= 226.6kbits/s dup=10004 drop=0 speed=1.01x    ' } +251ms
  cypress:server:video capture stderr log { message: 'frame=19244 fps= 25 q=12.0 size=   21248kB time=00:12:48.80 bitrate= 226.4kbits/s dup=10012 drop=0 speed=1.01x    ' } +582ms
  cypress:server:video capture stderr log { message: 'frame=19259 fps= 25 q=14.0 size=   21248kB time=00:12:49.40 bitrate= 226.2kbits/s dup=10020 drop=0 speed=1.01x    ' } +584ms
  cypress:server:video capture stderr log { message: 'frame=19273 fps= 25 q=12.0 size=   21248kB time=00:12:49.96 bitrate= 226.1kbits/s dup=10027 drop=0 speed=1.01x    ' } +583ms
  cypress:server:video capture stderr log { message: 'frame=19286 fps= 25 q=12.0 size=   21504kB time=00:12:50.48 bitrate= 228.6kbits/s dup=10034 drop=0 speed=1.01x    ' } +500ms
  cypress:server:video capture stderr log { message: 'frame=19298 fps= 25 q=12.0 size=   21504kB time=00:12:50.96 bitrate= 228.5kbits/s dup=10040 drop=0 speed=1.01x    ' } +500ms
  cypress:server:video capture stderr log { message: 'frame=19313 fps= 25 q=12.0 size=   21504kB time=00:12:51.56 bitrate= 228.3kbits/s dup=10048 drop=0 speed=1.01x    ' } +583ms
  cypress:server:video capture stderr log { message: 'frame=19325 fps= 25 q=12.0 size=   21504kB time=00:12:52.04 bitrate= 228.2kbits/s dup=10054 drop=0 speed=1.01x    ' } +500ms
  cypress:server:video capture stderr log { message: 'frame=19338 fps= 25 q=12.0 size=   21504kB time=00:12:52.56 bitrate= 228.0kbits/s dup=10061 drop=0 speed=1.01x    ' } +499ms
  cypress:server:video capture stderr log { message: 'frame=19352 fps= 25 q=12.0 size=   21504kB time=00:12:53.12 bitrate= 227.9kbits/s dup=10068 drop=0 speed=1.01x    ' } +585ms
  cypress:server:video capture stderr log { message: 'frame=19365 fps= 25 q=12.0 size=   21504kB time=00:12:53.64 bitrate= 227.7kbits/s dup=10075 drop=0 speed=1.01x    ' } +500ms
```

**Comment by denislutz** [comment]
Dear Cypress team, I love your work and cypress, but please can you resolve this issue, we dont upgrade because we fear that it comes again we are stuck with <8.0 version. Here is the topic on overflow: https://stackoverflow.com/questions/69658152/cypress-test-runner-unexpectedly-exited-via-a-exit-event-with-signal-sigsegv-in/70304003?noredirect=1#comment124359414_70304003

**Comment by davidlukerice** [comment]
Just ran into a similar issue when upgrading from cypress `v8.7` to `v9.5.3`. We run cypress headless in jenkins/aws as part of our CI process with
```
Cypress: 9.5.3
Browser: Electron 94
Node Version: v12.22.8
```
The only thing that ended up solving the issue for us was upgrading from a `c5x.large` to a `c5.2xlarge` to give cypress a bunch more ram and cpu power. Though, that seems overkill for something like this.

**Comment by waltz** [comment]
@davidlukerice Wow, that is a huge resource increase. Thanks for the heads up. Did you get any clues on what process was running out of memory?

**Comment by davidlukerice** [comment]
@waltz here's one of the `process_profiler` logs around the time of the timeout. It mostly looks like cypress and our plugins are fighting for resources, but didn't dig too much further into it. The only real plugins we're using are `@cypress/react/plugins/react-scripts` and a `dsw` services worker for mocking requests (these are for react component tests run via `NODE_ENV=test cypress run-ct` btw).
Turning of the tests that used the `dsw` service worker got the tests fully running again, but I'm not sure if it's `dsw` pushing it over the edge or just having more tests.
```
┌─────────┬───────────────────┬──────────────┬──────────────────────────────┬────────────┬────────────────┬──────────┬──────────────┬─────────────┐
│ (index) │       group       │ processCount │             pids             │ cpuPercent │ meanCpuPercent │ memRssMb │ meanMemRssMb │ maxMemRssMb │
├─────────┼───────────────────┼──────────────┼──────────────────────────────┼────────────┼────────────────┼──────────┼──────────────┼─────────────┤
│    0    │     'plugin'      │      3       │    '27010, 27030, 27037'     │   27.95    │     27.35      │ 1992.63  │   1496.14    │   1992.63   │
│    1    │     'cypress'     │      1       │           '26342'            │   51.36    │     35.45      │  265.66  │    255.08    │    270.2    │
│    2    │ 'electron-shared' │      4       │ '26350, 26515, 26351, 26534' │    7.85    │      5.59      │  234.49  │    214.05    │   234.49    │
│    3    │    'Electron'     │      1       │           '27194'            │    0.89    │      0.98      │  114.58  │    118.95    │   153.33    │
│    4    │     'ffmpeg'      │      1       │           '27185'            │   11.43    │      8.87      │  106.3   │    96.18     │    106.3    │
│    5    │      'other'      │      2       │        '29214, 29215'        │     0      │       0        │   3.49   │    11.71     │    94.22    │
│    6    │      'TOTAL'      │      12      │             '-'              │   99.48    │     71.48      │ 2717.15  │   1880.97    │   2717.15   │
└─────────┴───────────────────┴──────────────┴──────────────────────────────┴────────────┴────────────────┴──────────┴──────────────┴─────────────┘ 
```

**Comment by shdb2** [comment]
I experience something similar in my GitLab pipeline. The behavior seems to be a bit random but often enough to be a major problem:

>   Cypress:        9.5.4                                                                          
>   Browser:        Chrome 97 (headless)                                                          
>   Node Version: v16.13.2 (/usr/local/bin/node)                                                
>   Specs:          3 found (...)     

> System Platform: linux (Debian - 10.11)
> System Memory: 8.05 GB free 2.16 GB
                     
The error message:

>  CypressError: Cypress command timeout of 4000ms exceeded.
> ...."after each" hook for "some test case name":
     CypressError: Cypress detected that you returned a promise from a command while also invoking one or more cy commands in that promise.`
The command that returned the promise was:
  > `cy.click()`
The cy command you invoked inside the promise was:
  > `cy.log()`
Because Cypress commands are already promise-like, you don't need to wrap them or return your own promise.
Cypress will resolve your command with whatever the final Cypress command yields.
The reason this is an error instead of a warning is because Cypress internally queues commands serially whereas Promises execute as soon as they are invoked. Attempting to reconcile this would prevent Cypress from ever resolving.
https://on.cypress.io/returning-promise-and-commands-in-another-command
Because this error occurred during a `after each` hook we are skipping all of the remaining tests.`

In my tests, there is not a single `afterEach` or `afterAll` method, so that error is really confusing. The test itself is really simple and does not rely on external services.

Retrying the tests in the pipeline might lead to a successful execution sometimes. If I disable the video recording, the tests are (for now?) always successful. So it might be related to the videos that are recorded after each test suite.

**Comment by sk25469** [comment]
I was using the `cypress-included-9.7.0` in docker and running tests inside it. It had `chrome version 104`. Some of the tests were getting stuck as mentioned in this issue https://github.com/cypress-io/cypress/issues/22506.
I tried disabling the cypress command logs but the issue was still there. Turns out this changing the `chrome version < 100` was the actual solution in my case. I changed to `cypress-included-8.7.0` which had the `chrome version 91` and the tests ran successfully without any issues. Then i tested it for `chrome version 95` and it also ran without issues there.
